ContextApp/src/main/java/com/github/migi_1/ContextApp/HeartsUpdateFunctions.java

```diff
@@ -0,0 +1,136 @@
+package com.github.migi_1.ContextApp;
+
+import android.widget.ImageView;
+import com.github.migi_1.ContextMessages.PlatformPosition;
+
+/**
+ * Contains functions which changes the hearts that represent the 
+ * player's health.
+ */
+public class HeartsUpdateFunctions {
+
+    private MainActivity act;
+    private PlatformPosition position;
+    
+    /**
+     * This creates an instance of the HeartsUpdateFunctions.
+     * @param act
+     * 			The instance of the application that calls this function
+     * @param position 
+     *                  The position of the player whose hearts will be updated
+     */
+    public HeartsUpdateFunctions(MainActivity act, PlatformPosition position) {
+        this.act = act;
+        this.position = position;
+    }
+    
+   /**
+    * Calls functions to make the hearts the right colour.
+    * @param health the amount of health that has to be displayed in grey
+    *      and red hearts.
+    */
+    public void setHealth(int health) {
+        
+        switch (health) {
+            case 0:
+                makeGrey(1);
+                makeGrey(2);
+                makeGrey(3);
+                break;
+            case 1:
+                makeRed(1);
+                makeGrey(2);
+                makeGrey(3);
+                break;
+            case 2:
+                makeRed(1);
+                makeRed(2);
+                makeGrey(3);
+                break;
+            default:
+                makeRed(1);
+                makeRed(2);
+                makeRed(3);
+                break;
```

Comment:
It seems that your takeDamage method allows negative health. If this happens you will enter the default case, which is probably not intended